evennia/utils/utils.py

```diff
@@ -2160,3 +2161,120 @@ def decorator(*args, **kwargs):
             return ret
 
     return decorator
+
+
+class FunctionArgument(object):
+    def __init__(self, name):
+        self.name = name.strip()
+
+    def __str__(self):
+        return self.name
+
+
+def parse_arguments(s, **kwargs):
+    """
+    This method takes a string and parses it as if it were an argument list to a function.
+    It supports both positional and named arguments.
+
+    Values are automatically converted to int or float if possible.
+    Values surrounded by single or double quotes are treated as strings.
+    Any other value is wrapped in a "FunctionArgument" class for later processing.
+
+    Args:
+        s (str): The string to convert.
+
+    Returns:
+        (list, dict): A tuple containing a list of arguments (list) and named arguments (dict).
+    """
+    global _ARG_ESCAPE_SIGN
+
+    args_list = []
+    args_dict = {}
+
+    # State (general)
+    inside = (False, None)  # Are we inside a quoted string? What is the quoted character?
+    skip = False  # Skip the current parameter?
+    escape = False  # Was the escape key used?
+    is_string = False  # Have we been inside a quoted string?
+    temp = ""  # Buffer
+    key = None  # Key (for named parameter)
+
+    def _parse_value(temp):
+        ret = temp.strip()
+        if not is_string:
+            try:
+                ret = int(ret)
+            except ValueError:
+                try:
+                    ret = float(ret)
+                except ValueError:
+                    if ret != "":
+                        return FunctionArgument(ret)
+
+        return ret
```

Comment:
1. > Flat is better than nested.
[PEP 20 -- Zen of Python](https://www.python.org/dev/peps/pep-0020/)
3. Do not use the same name for local variable and outer scope variables. Put one trailing underscore to avoid confusion. It can lead to the misbelief that attributing value to these variables change the variable from outer scope.

```suggestion
    def _parse_value(temp_):
        ret = temp_.strip()
        if is_string:
            return ret
        try:
            return int(ret)
        except ValueError:
            pass
        try:
            return float(ret)
        except ValueError:
            pass
        return ret if ret == "" else None
```